src/main/java/Duke.java

```diff
@@ -1,10 +1,96 @@
+import java.util.Scanner;
 public class Duke {
+    public static void listTasks(Task[] list, int i){
+        System.out.println("____________________________________________________________");
+        for (int j = 0; j < i; j++) {
+            System.out.println(Integer.toString(j + 1) + list[j]);
+        }
+        System.out.println("____________________________________________________________");
+    }
+    public static void updateMarkTask(Task[] list, int i, String line, boolean mark){
+        int index;
+        try{
+            index = Integer.parseInt(line.split(" ")[1]) - 1;
+        }catch (NumberFormatException e){
+            System.out.println("____________________________________________________________");
+            System.out.println(" Error: Invalid index (Not an integer).");
+            System.out.println("____________________________________________________________");
+            return;
+        }
+        if (index >= i){
+            System.out.println("____________________________________________________________");
+            System.out.println(" Error: Invalid index (Out of bounds).");
+            System.out.println("____________________________________________________________");
+            return;
+        }
+        System.out.println("____________________________________________________________");
+        if (mark){
+            System.out.println(" Nice! I've marked this task as done:");
+            System.out.println("  [X] " + list[index].getName());
+        }else{
+            System.out.println(" OK, I've marked this task as not done yet:");
+            System.out.println("  [ ] " + list[index].getName());
+        }
+        System.out.println("____________________________________________________________");
+        list[index].setMarked(mark);
+    }
+    public static int addNewTask(Task[] list, int i, String line){
```

Comment:
Not sure how I feel about combining the functionality of multiple commands into the same function, although I can see your reasoning given that they do very similar things.